File: Try_GameEngine5/src/com/example/try_gameengine/utils/DetectAreaRequest.java
package com.example.try_gameengine.utils;

/**
 * {@code DetectAreaRequest} is an concurrent class by implements {@code IDetectAreaRequest}.
 * 
 * @author irons
 * 
 */
public class DetectAreaRequest implements IDetectAreaRequest {
	private DetectArea detectArea;

	/**
	 * @param request
	 */
	public DetectAreaRequest(DetectArea request) {
		this.detectArea = request;
	}

	@Override
	public void setDetectArea(DetectArea request) {
		this.detectArea = request;
	}

	@Override
	public DetectArea getDetectArea() {
		return detectArea;
	}

	@Override
	public String getTag() {
		return detectArea.getTag();
	}

	@Override
	public void setTag(String tag) {
		detectArea.setTag(tag);
	}

	@Override
	public Object getObjectTag() {
		return detectArea.getObjectTag();
	}

	@Override
	public void setObjectTag(Object objectTag) {
		detectArea.setObjectTag(objectTag);
	}
}

File: Try_GameEngine5/src/com/example/try_gameengine/utils/DetectAreaRequest.kt
package com.example.try_gameengine.utils

/**
 * `DetectAreaRequest` is an concurrent class by implements `IDetectAreaRequest`.
 * 
 * @author irons
 // */
class DetectAreaRequest(request: DetectArea) : IDetectAreaRequest {
    private var detectArea: DetectArea

    /**
     * @param request
     // */
    init {
        this.detectArea = request
    }

    override fun setDetectArea(request: DetectArea) {
        this.detectArea = request
    }

    override fun getDetectArea(): DetectArea {
        return detectArea
    }

    override fun getTag(): String? {
        return detectArea.getTag()
    }

    override fun setTag(tag: String?) {
        detectArea.setTag(tag)
    }

    override fun getObjectTag(): Any? {
        return detectArea.getObjectTag()
    }

    override fun setObjectTag(objectTag: Any?) {
        detectArea.setObjectTag(objectTag)
    }
}
